### ISSUE TITLE
Inconsistent S3 bucket prefix

### ISSUE THREAD

**Issue by c-knowles** [kubernetes-incubator/kube-aws#334]
While running a validate using v0.9.4-rc.3 I noticed the main stack appears at `${S3_URI}/${CLUSTER_NAME}` but the sub-stacks appear at `${S3_URI}/kube-aws/clusters/${CLUSTER_NAME}/exported/stacks`. Shall we make this consistent? In my case I'm sometimes doing `aws s3 sync` to grab those files back to have a look at them and ideally it'd be good for that to just go into one S3 prefix.

My workaround is currently to put the cluster name in the S3 URI so we end up with:
```
${S3_BASE}/${CLUSTER_NAME_FROM_S3_URI}/${CLUSTER_NAME}
${S3_BASE}/${CLUSTER_NAME_FROM_S3_URI}/kube-aws/clusters/${CLUSTER_NAME}/exported/stacks
```
But at least then everything related to one cluster can be found in a single place.

**Comment by c-knowles** [comment]
Thanks for the quick turnaround on this. I used v0.9.4 release (not RC) and the problem seems to persists:

1. Cleared out S3 bucket entirely, zero items
2. Run `kube-aws validate --s3-uri s3://${BUCKET_NAME}/${CLUSTER_NAME_FROM_S3_URI}`
3. Run `aws s3 sync s3://${BUCKET_NAME}/${CLUSTER_NAME_FROM_S3_URI} tmp/uploaded-cf/`

The result is a structure like this inside `tmp/uploaded-cf/`:
```
${CLUSTER_NAME_FROM_S3_URI}/${CLUSTER_NAME}/stack.json
${CLUSTER_NAME_FROM_S3_URI}/kube-aws/clusters/${CLUSTER_NAME}/exported/stacks/control-plane/stack.json
${CLUSTER_NAME_FROM_S3_URI}/kube-aws/clusters/${CLUSTER_NAME}/exported/stacks/worker-pool-a/stack.json
... (other worker pools)
```

**Comment by c-knowles** [comment]
I just noticed on an `kube-aws update` with the same S3 URI I now get this extra file:

```
${CLUSTER_NAME_FROM_S3_URI}/kube-aws/clusters/${CLUSTER_NAME}/exported/stacks/${CLUSTER_NAME}/stack.json
```

So actually I think a couple of things need to happen:
1. Move this file to `/kube-aws/clusters/${CLUSTER_NAME}/stack.json` (`update` command, maybe others too)
1. Ensure `/${CLUSTER_NAME}/stack.json` is also moved to here (`validate` command)

**Comment by mumoshu** [comment]
Thanks for the detailed report and the test 👍 I'll tackle this

**Comment by mumoshu** [comment]
Btw, `/${CLUSTER_NAME}/stack.json` and `/kube-aws/clusters/${CLUSTER_NAME}/stack.json` are redundant (should not exist) and all of them should be merged into `${CLUSTER_NAME_FROM_S3_URI}/kube-aws/clusters/${CLUSTER_NAME}/exported/stacks/${CLUSTER_NAME}/stack.json`, which is meant for the root stack!

**Comment by mumoshu** [comment]
@c-knowles The fix is in #465. Things are much simpler now.

* `${CLUSTER_NAME_FROM_S3_URI}/${CLUSTER_NAME}/stack.json` is not created anymore
* `${CLUSTER_NAME_FROM_S3_URI}/kube-aws/clusters/${CLUSTER_NAME}/exported/stacks/${CLUSTER_NAME}/stack.json` is created on `kube-aws validate/update/up`.

**Comment by c-knowles** [comment]
Tested against latest master and seems to be working. Probably would have called the second `${CLUSTER_NAME}` folder just `root` to match the naming in kube-aws but at least everything is under one prefix now so the original report is solved.

For anyone checking here, everything is now under `${S3_URI}/kube-aws/clusters/${CLUSTER_NAME}/exported/stacks/`
